### DISCUSSION TITLE
Build desktop app from source

### DISCUSSION THREAD

**Discussion by simisimis** [TriliumNext/Trilium#2717] [General]
Hi,
I am struggling to find any information on how do I build linux desktop application from source instead of using provided binaries.

**Comment by Nriver** [comment]
There are some build scripts in the `bin` directory of this repo. https://github.com/zadam/trilium/tree/master/bin
I never tried them before. Maybe you can have a look at them.

**Reply by simisimis** [reply]
thanks, that looks exactly what I was hoping to find, did not expect that stuff to be in bin 😅

**Reply by simisimis** [reply]
although in the scripts I see stuff being copied/moved around while I need to recompile things as my linux distro(nixos) is non FHS and many expected paths like /usr or /bin don't exist. but I will check another day if I can salvage something from there. thx again

**Reply by Nriver** [reply]
Good luck！
